### PR TITLE
Fix iterator drag-in & releasing

### PR DESCRIPTION
**Pull request by joeyballentine**
what we talked about in discord

### CODE REVIEW

File: src/renderer/components/ReactFlowBox.tsx [hunk 1]
```diff
@@ -255,14 +255,18 @@ export const ReactFlowBox = memo(({ wrapperRef, nodeTypes, edgeTypes }: ReactFlo
...
+                            ...edges.filter((e) => {
+                                if (e.source === node.id) {
+                                    const target = nodes.find((n) => n.id === e.target);
+                                    if (target) {
+                                        return target.parentNode !== iterInBounds.id;
+                                    }
+                                }
+                                return false;
+                            }),
+                        ];
```
Comment (RunDevelopment) [inline src/renderer/components/ReactFlowBox.tsx:269]: While I commend your commitment to immutable data structures, using `push` would probably be easier.
```suggestion
                        edgesToRemove.push(
                            ...edges.filter((e) => {
                                if (e.source === node.id) {
                                    const target = nodes.find((n) => n.id === e.target);
                                    if (target) {
                                        return target.parentNode !== iterInBounds.id;
                                    }
                                }
                                return false;
                            }),
                        );
```

File: src/renderer/contexts/GlobalNodeState.tsx [hunk 2]
```diff
@@ -582,10 +582,14 @@ export const GlobalProvider = memo(
 
         const releaseNodeFromParent = useCallback(
             (id: string) => {
+                let parentNode: Node<NodeData> | undefined;
+                let node: Node<NodeData> | undefined;
+                let capturedNodes: Node<NodeData>[];
```
Comment (RunDevelopment) [inline src/renderer/contexts/GlobalNodeState.tsx:587]: You assumed that state updates will be processed in order. While this guaranteed for a single memory cell, I don't think it's guaranteed for different memory cells. `changeNodes` and `changeEdges` ultimately use `useNodesState`'s and `useEdgesState`'s setters, which probably use different cells.

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

File: src/renderer/contexts/GlobalNodeState.tsx (hunk 2 above)
Comment (joeyballentine) [reply src/renderer/contexts/GlobalNodeState.tsx:587]: so what do you recommend I do here? use getNodes in the changeEdges callback?

Comment (RunDevelopment) [reply src/renderer/contexts/GlobalNodeState.tsx:587]: I'm pretty sure using `getNodes` inside a state update isn't defined either. I think we need to `getNodes()` and figure out what edges to remove before `changeEdges`.

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
This fixes the 2 bugs, but I found another one. If you drag multiple nodes into an iterator, all edges between them will be removed.

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**